components/brave_news/browser/feed_fetcher.cc

```diff
@@ -152,47 +157,76 @@ void FeedFetcher::OnFetchFeedFetchedFeed(
     return;
   }
 
-  std::move(callback).Run({locale, etag, ParseFeedItems(result.value_body())});
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      base::BindOnce([](base::Value value) { return ParseFeedItems(value); },
+                     std::move(result.value_body())),
+      base::BindOnce(
+          [](std::string locale, std::string etag,
+             FetchFeedSourceCallback callback,
+             std::vector<mojom::FeedItemPtr> items) {
+            std::move(callback).Run(
+                {std::move(locale), std::move(etag), std::move(items)});
+          },
+          std::move(locale), std::move(etag), std::move(callback)));
```

Comment:
Is it safe to call `callback` later?

Reply:
I've updated these to check a weakptr before running the callback.